db/sql/00_msar.sql

```diff
@@ -2294,29 +2294,38 @@ $$ LANGUAGE SQL;
 
 
 CREATE OR REPLACE FUNCTION
-msar.add_mathesar_table(sch_oid oid, tab_name text, col_defs jsonb, con_defs jsonb, comment_ text)
+msar.add_mathesar_table(sch_id oid, tab_name text, col_defs jsonb, con_defs jsonb, comment_ text)
   RETURNS oid AS $$/*
 Add a table, with a default id column, returning the OID of the created table.
 
 Args:
-  sch_oid: The OID of the schema where the table will be created.
-  tab_name: The unquoted name for the new table.
+  sch_id: The OID of the schema where the table will be created.
+  tab_name (optional): The unquoted name for the new table.
   col_defs (optional): The columns for the new table, in order.
   con_defs (optional): The constraints for the new table.
   comment_ (optional): The comment for the new table.
 
-Note that even if col_defs is null, we will still create a table with a default 'id' column. Also,
+Note that if tab_name is null, the table will be created with a name in the format 'Table <n>'.
+If col_defs is null, the table will still be created with a default 'id' column. Also,
 if an 'id' column is given in the input, it will be replaced with our default 'id' column. This is
 the behavior of the current python functions, so we're keeping it for now. In any case, the created
 table will always have our default 'id' column as its first column.
 */
 DECLARE
+  schema_name text;
+  table_count integer;
   fq_table_name text;
   created_table_id oid;
   column_defs __msar.col_def[];
   constraint_defs __msar.con_def[];
 BEGIN
-  fq_table_name := format('%I.%I', msar.get_schema_name(sch_oid), tab_name);
+  schema_name := msar.get_schema_name(sch_id);
+  IF NULLIF(tab_name, '') IS NOT NULL THEN
+    fq_table_name := format('%I.%I', schema_name, tab_name);
+  ELSE
+    SELECT COUNT(*) INTO table_count FROM pg_catalog.pg_class WHERE relkind = 'r' AND relnamespace = sch_id;
+    fq_table_name := format('%I.%I', schema_name, 'Table ' || (table_count + 1));
```

Comment:
Correct me if I'm wrong, but I think there may be a bug here. Say I begin with zero tables...

1. I create a table. It gets named `Table 1`.
1. I create another table. It gets named `Table 2`.
1. I delete the first table.
1. I create a third table. Within this function `table_count` will be `1` so `fq_table_name` will be something like `public."Table 2"`. But `public."Table 2"` already exists so `CREATE TABLE` will fail.

Right?

Reply:
You are correct @seancolsen, I've fixed the bug and added a test in 0450fbe, thanks for catching it!

Reply:
Cool. I haven't reviewed thoroughly, but your changes look good from a glance. I'll leave it to Brent to review fully.